management/src/main/scala/akka/management/javadsl/HealthChecks.scala

```diff
@@ -30,16 +29,16 @@ final class HealthChecks(system: ExtendedActorSystem, settings: HealthCheckSetti
   /**
    * Returns CompletionStage(true) if the system is ready to receive user traffic
    */
-  def ready(): CompletionStage[java.lang.Boolean] =
-    delegate.ready().map(Boolean.box)(ExecutionContexts.sameThreadExecutionContext).toJava
+  def ready(): CompletionStage[Either[String, Unit]] =
```

Comment:
we shouldn't use `Either` in javadsl

Reply:
Good point - ideas on what a good alternative would be for Java (if this PR goes forward)? Maybe `Optional<String>` (plus delegated methods requested by @chbatey)?

Reply:
Probably a custom type is more idiomatic in Java